integration/tests/cook/util.py

```diff
@@ -1163,10 +1163,12 @@ def slave_cpus(mesos_url, hostname):
     return slave_cpus
 
 
-def slave_pool(mesos_url, hostname):
+def slave_pool(cook_url, mesos_url, hostname):
     """Returns the pool of the specified Mesos agent, or None if the agent doesn't have the attribute"""
     slaves = get_mesos_state(mesos_url)['slaves']
-    pool = next(s.get('attributes', {}).get('cook-pool', None) for s in slaves if s['hostname'] == hostname)
+    pool_selection_plugin_config = settings(cook_url).get('plugins', {}).get('pool-selection', {})
+    attribute_name = pool_selection_plugin_config.get('attribute-name', None) or 'cook-pool'
+    pool = next(s.get('attributes', {}).get(attribute_name, None) for s in slaves if s['hostname'] == hostname)
```

Comment:
Might be worth extracting a function since we have the same code in `hosts_to_consider`

Reply:
Done